src/account/AccountStorage.sol

```diff
@@ -63,8 +63,14 @@ struct AccountStorage {
     mapping(IPlugin => mapping(address => PermittedExternalCallData)) permittedExternalCalls;
     // For ERC165 introspection
     mapping(bytes4 => uint256) supportedIfaces;
+    // Installed plugins capable of signature validation.
+    EnumerableSet.Bytes32Set signatureValidations;
 }
 
+// TODO: Change how pre-validation hooks work to allow association with validation, rather than selector.
+// Pre signature validation hooks
+// mapping(FunctionReference => EnumerableSet.Bytes32Set) preSignatureValidationHooks;
```

Comment:
Curious to understand the typical use cases for preHooks in signature validations.

Reply:
I think the simple ones are performing checks like limiting who can validate something - i.e. a session key that is allowed to sign `permit` messages, but only for USDC and only to a known spender.

Reply:
Hmm, when the account gets a 1271 call, we only receive a `bytes32 hash` of all the input parameters, we'll need to get the struct/type + rebuild it to be able to check against the struct parameters

Reply:
There are 3 main dimensions of association to consider here - the validation plugin, the typehash, and the address of the app